### PR TITLE
Add more schema validations for select operator

### PR DESCRIPTION
**Pull request by nonibansal**

### CODE REVIEW

File: fennel/datasets/datasets.py [hunk 1]
```diff
@@ -2770,12 +2796,46 @@ def visitDrop(self, obj) -> DSSchema:
...
+        key_fields = input_schema.keys
+
+        output_schema_name = f"'[Pipeline:{self.pipeline_name}]->select node'"
+        if obj.columns is None or len(obj.columns) == 0:
+            raise ValueError(
+                f"invalid select - {output_schema_name} must have at least one column to select"
+            )
+        if timestamp not in obj.columns:
+            raise ValueError(
+                f"invalid select - {output_schema_name} timestamp_field : `{timestamp}` must be in columns"
```
Comment (nikhilgarg28) [inline fennel/datasets/datasets.py:2822]: Super nit but can you please remove the underscore from `timestamp_field` and make it two words?

File: fennel/datasets/datasets.py [hunk 2]
```diff
@@ -2770,12 +2796,46 @@ def visitDrop(self, obj) -> DSSchema:
...
+            )
+        for key_field in key_fields:
+            if key_field not in obj.columns:
+                raise ValueError(
+                    f"invalid select - {output_schema_name} key_field : `{key_field}` must be in columns"
```
Comment (nikhilgarg28) [inline fennel/datasets/datasets.py:2827]: Same here - key_field -> `key field`

**Review by nikhilgarg28 — APPROVED** [review]

File: fennel/datasets/datasets.py (hunk 1 above)
Comment (nonibansal) [reply fennel/datasets/datasets.py:2822]: changed

File: fennel/datasets/datasets.py (hunk 2 above)
Comment (nonibansal) [reply fennel/datasets/datasets.py:2827]: changed

**Pull request merged**